objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+                                end
+                            end
+                        else
+                            if tbl.container then
+                                Player[params.color].broadcast("Cannot find impended card: " .. tbl.name .. "\nThis is in bag or deck with guid: " .. tbl.container .. "\nIf expected, ignore this message. Otherwise, please find it and put it in play!", Color.Orange)
+                            else
+                                Player[params.color].broadcast("Cannot find impended card: " .. tbl.name .. "\nIf expected, ignore this message. Otherwise, please find it and put it in play!", Color.Orange)
+                            end
+                        end
+                    end
+                end
+                -- Place all ready cards into play
+                local spos = dances.getPosition()
+                local xOffset = -9.5
+                local xScale = 19
+                local xSpacing = 1/math.max(5, #toPlayFast + #toPlaySlow - 1)
+                local yOffset = -0.22
+                local ySpacing = 0.05
+                local zOffset = 16
+                for ind,card in pairs(toPlayFast) do
+                    card.use_hands = false
+                    card.setPositionSmooth(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset), false, false)
+                    card.setRotation({0,180,0})
+                    Wait.condition(
+                        function() card.use_hands = true end,
+                        function() return not card.isSmoothMoving() end,
+                        3, -- don't think this timeout function will be necessary, but thought I'd add it to be safe
+                        function()
+                            card.use_hands = true
+                            card.setPosition(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset))
+                        end
+                    )
+                end
+                for ind,card in pairs(toPlaySlow) do
+                    card.use_hands = false
+                    card.setPositionSmooth(spos + Vector(xOffset+xScale*(ind+#toPlayFast-1)*xSpacing, yOffset+(ind+#toPlayFast-1)*ySpacing, zOffset), false, false)
+                    card.setRotation({0,180,0})
+                    Wait.condition(
+                        function() card.use_hands = true end,
+                        function() return not card.isSmoothMoving() end,
+                        3, -- don't think this timeout function will be necessary, but thought I'd add it to be safe
+                        function()
+                            card.use_hands = true
+                            card.setPosition(spos + Vector(xOffset+xScale*(ind-1)*xSpacing, yOffset+(ind-1)*ySpacing, zOffset))
+                        end
+                    )
+                end
+            end
+        end
+    end
+end
+
+function onObjectEnterContainer(cont, obj)
```

Comment:
what's the use case for an impended card to be in a container?

Reply:
Stacking two cards creates a deck container, and it's not too hard to accidentally stack a couple of cards.

Reply:
Either that or since you can forget a card that is impended, a player may not remember to reset an impended card before they forget and this both prevents any errors occurring from cards not being able to be found while also giving the ability to get info about it. I did consider something like it sending a chat message that you can click on to pull the card out if it was a mistake, but that felt a little overkill.

Reply:
sounds like adding a hook for when a power card is forgotten could be helpful here?